File: Leetcode/Leetcode10/resultArray.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var resultArray = function(nums) {
    const nums1 = [], nums2 = [];
    const iterator = nums[Symbol.iterator]();
    nums1.push(iterator.next().value);
    nums2.push(iterator.next().value);
    while (true) {
        const temp = iterator.next();
        if (temp.done) break;
        if (nums1.at(-1) > nums2.at(-1)) {
            nums1.push(temp.value);
        } else {
            nums2.push(temp.value);
        }
    }
    return [...nums1, ...nums2];
};

File: Leetcode/Leetcode10/resultArray.cpp
#include <vector>

using namespace std;

class Solution {
public:
    vector<long long> resultArray(vector<int>& nums, int k) {
        vector<vector<long long>> dp(nums.size(), vector<long long>(k, 0));
        for (int i = 0; i < nums.size(); i++) {
            dp[i][nums[i] % k]++;
            for (int j = 0; j < k; j++) {
                if (i - 1 >= 0) {
                    dp[i][(long long)nums[i] * j % k] += dp[i - 1][j];
                }
            }
        }

        vector<long long> ans(k, 0);
        for (int i = 0; i < k; i++) {
            for (int j = 0; j < nums.size(); j++) {
                ans[i] += dp[j][i];
            }
        }
        return ans;
    }
};